nav2_loopback_sim/src/loopback_simulator.cpp

```diff
@@ -0,0 +1,227 @@
+
+    // Publish identity transform between odom and base_footprint
+    publishTransform(zero_pose, "odom", "base_footprint");
+
+    relocalization_pose_ = init_pose_.pose.pose;
+    zero_odom_base_published_ = true;
+    return;
+  }
+
+  // Updating map->base_link transform based on set pose
+  geometry_msgs::msg::TransformStamped map_to_base_link_transform;
+  map_to_base_link_transform.header = init_pose_.header;
+  map_to_base_link_transform.header.frame_id = "map";
+  map_to_base_link_transform.child_frame_id = "base_link";
+  map_to_base_link_transform.transform.translation.x = init_pose_.pose.pose.position.x;
+  map_to_base_link_transform.transform.translation.y = init_pose_.pose.pose.position.y;
+  map_to_base_link_transform.transform.translation.z = init_pose_.pose.pose.position.z;
+  map_to_base_link_transform.transform.rotation = init_pose_.pose.pose.orientation;
+
+  geometry_msgs::msg::TransformStamped odom_to_base_link_transform;
+  try {
+    odom_to_base_link_transform = tf_buffer_->lookupTransform(
+      "odom",
+      "base_link",
+      tf2::TimePointZero);
+  } catch (tf2::TransformException & ex) {
+    RCLCPP_ERROR(this->get_logger(), "%s", ex.what());
+    return;
+  }
+
+  // Calculating map->odom transform
+  geometry_msgs::msg::TransformStamped map_to_odom_transform;
+  tf2::doTransform(map_to_base_link_transform, map_to_odom_transform, odom_to_base_link_transform);
+
+  // Updating relocalization pose of odom frame
+  relocalization_pose_.position.x = map_to_odom_transform.transform.translation.x;
+  relocalization_pose_.position.y = map_to_odom_transform.transform.translation.y;
+  relocalization_pose_.position.z = map_to_odom_transform.transform.translation.z;
+  relocalization_pose_.orientation = map_to_odom_transform.transform.rotation;
+
+  // Publishing updated position of odom and base_link wrt map
+  publishTransform(relocalization_pose_, "map", "odom");
+  publishTransform(init_pose_.pose.pose, "map", "base_link");
+}
+
+void LoopbackSimulator::timerCallback()
+{
+  if (!init_pose_set_) {
+    return;
+  }
+
+  // RCLCPP_INFO(this->get_logger(), "Publishing map->odom tf");
+
+  // Continously publishing map->odom transform
+  publishTransform(relocalization_pose_, "map", "odom");
+
+  // publish bogus lidar data
+  auto scan_msg = std::make_shared<sensor_msgs::msg::LaserScan>();
+
+  scan_msg->header.frame_id = "base_link";
+  scan_msg->header.stamp = this->now();
+
+  // Fill in the scan data
+  scan_msg->angle_min = -M_PI / 2.0;
+  scan_msg->angle_max = M_PI / 2.0;
+  scan_msg->angle_increment = M_PI / 180.0;
+  scan_msg->time_increment = 0.0;
+  scan_msg->scan_time = 0.1;
+  scan_msg->range_min = 0.0;
+  scan_msg->range_max = 10.0;
+
+  // Fill in the scan data
+  int num_readings = 180;
+  scan_msg->ranges.resize(num_readings);
+  for (int i = 0; i < num_readings; ++i) {
+    scan_msg->ranges[i] = 5.0;  // Set all ranges to 5 meters
+  }
+
+  scan_publisher_->publish(*scan_msg);
```

Comment:
```suggestion
  scan_publisher_->publish(std::move(scan_msg));
```

Once its a unique pointer